jee-learning-platform/backend/app/routes/ai.py

```
@@ -0,0 +1,356 @@
+            
+            if is_correct:
+                correct_answers += 1
+            
+            total_time += time_taken
+            
+            # Create checkpoint-like tracking for quiz questions
+            # This helps in mastery calculation
+            topic_id = question.get("topic_id")
+            if topic_id:
+                # Create a temporary checkpoint record for analytics
+                quiz_answer = models.UserAnswer(
```

Ruff (0.11.9)

Local variable `quiz_answer` is assigned to but never used

Remove assignment to unused variable `quiz_answer`

(F841)